docs/EVALUATION.md

```diff
@@ -0,0 +1,372 @@
+
+Similar windowing strategies as in the post-hoc offline analysis might yield useful results. These windows can limit the time range and sample count of evaluation data.
+
+### 3.2 Post-hoc offline analysis
+
+In experiment mode we can potentially utilize the full evaluation split over the whole dataset time range. Being able to "look ahead" is important to determine the performance of a model post factum as the usage period of model is after the training.
+
+#### Which dimensions are there? (controllable & observable)
+
+Naturally when running a pipeline we can a observe $m$ models being trained over time at the triggers $t_1, ..., t_i \in T$. We refer to $m_i$ as the $i$-th model trained after trigger $t_i$.
+
+Every of these $m$ models can be evaluated at any point in time (bounded by the dataset time range). As we are potentially interested in more than one evaluation metric $q_i \in Q$ ($Q$ being the set of evaluation metrics that are of interest to us), we can evaluate every model at every point in time with every evaluation metric. We therefore get a time series of evaluation results from $M \times Q$.
+
+#### 3.2.1 Analysis of a single pipeline run (comparing the different models)
+
+<details>
+  <summary><b>Plot 1: (line plot, x=time, y=eval metric, color=model)
+</b></summary>
+
+When analyzing how a single pipeline run performed over time, we can line-plot the evaluation results over time using time as the $x$ dimension, the evaluation metric value as the $y$ dimension and the model as a color coded dimension. The different evaluation metrics can be plotted in different plots using facets.
+
+Using this plot we expect to see that every model performs best at the interval it was trained in(, and hopefully also reasonably well in the interval the model was served in). Accuracy should decay in both directions from the training interval.
+
+There are two main approaches to acquire those time series of evaluation results:
+</details>
+
+##### Dataset slicing (`MatrixEvaluationStrategy`)
+
+Partitioning the dataset(s) w.r.t. to time at sufficiently small intervals will allow us to extract a time series after running evaluations for every (slice, model, metric) combination.
+
+When plotting this there are several cases:
+
+- When our slices already correspond to the minimal resolution of the samples times (e.g. every year), we don't need to think about interpolation and can just connect the measurement points with lines.
+
+- When we have more continuous data where we cannot evaluate at every point in time, can either plot vertical lines (same value for a certain time range) or interpolate. In the latter case it's fair to plot the points for each evaluation interval at the interval center and connect the points with lines (linear interpolation).
```

Comment:
One thing related to this I believe is the whole patching stuff. I wanted to discuss this with you anyways again. Even for SIGMOD, we still needed to patch the logs for CGLM, but not anymore for yearbook, if I remember correctly. There was this problem with the start timestamp being 0 and having triggers. We should think about the fundamental issue here again, and at least document it (also in the README of the datasets that are affected, but also here). Can we solve this somehow? It has been some weeks, I don't remember the details right now with all the other stuff going on, but in the best case, we could avoid patching at all. Maybe we can discuss this in our 1:1 this week also, if that's easier

Reply:
I thought we had those issues solved with the latest revision on main. However, when I rerun some of the experiments for my thesis (with the newer trigger variants), I will see if they are still present and potentially address them.

Reply:
@XianzheMa do you remember which pipelines still need to be patched? CGLM right?

Reply:
Okay, thanks for the reminder and the explanation! It would be good if one of you @MaxiBoether @XianzheMa can send me a recent pipeline log file so I can inspect the issue in detail.
Generally, I don't want to include these patching things in global documentation, but I want to fix the business logic so it doesn't require patching anymore. 
That's why I propose to merge this PR without mentioning patching.